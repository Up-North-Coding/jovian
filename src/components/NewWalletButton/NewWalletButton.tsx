import React from "react";
import Button from "@mui/material/Button";
import useAccount from "hooks/useAccount";

const NewWalletButton: React.FC = () => {
  const { accountRs, accountSeed, fetchFn } = useAccount();

  if (fetchFn === undefined) {
    return <></>;
  }

  if (!accountRs || !accountSeed) {
    return (
      <Button onClick={() => fetchFn()} variant="contained">
        New Wallet
      </Button>
    );
  }

  return (
    <>
      <p>Account: {accountRs}</p>
      <p>Seed: {accountSeed}</p>
    </>
  );
};

export default React.memo(NewWalletButton);
