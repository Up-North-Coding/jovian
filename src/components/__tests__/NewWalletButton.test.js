import React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";
import NewWalletButton from "components/NewWalletButton";

let windowSpy;
beforeEach(() => {
  // fixes an error during button render
  const { getComputedStyle } = window;
  window.getComputedStyle = (elt) => getComputedStyle(elt);

  window.crypto = { getRandomValues: jest.fn() };

  windowSpy = jest.spyOn(window.crypto, "getRandomValues");
});
// Note: running cleanup afterEach is done automatically for you in @testing-library/react@9.0.0 or higher
// unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

it("should take a click and generate a seed (eventually a wallet address)", () => {
  const { getByRole } = render(<NewWalletButton />);
  const button = getByRole("button", {
    name: /new wallet/i,
  });

  expect(button).toMatchSnapshot();

  fireEvent.click(button);
  expect(windowSpy).toHaveBeenCalledTimes(2);
});
