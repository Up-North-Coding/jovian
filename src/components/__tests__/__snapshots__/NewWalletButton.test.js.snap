// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a click and generate a seed (eventually a wallet address) 1`] = `
<button
  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  New Wallet
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
